backend/src/layer/data/database/mealplan_management.rs

```diff
@@ -0,0 +1,842 @@
+            ",
+            similar_name,
+            allergens
+                .iter()
+                .copied()
+                .map(Allergen::to_db_string)
+                .collect::<Vec<_>>() as _,
+            additives
+                .iter()
+                .copied()
+                .map(Additive::to_db_string)
+                .collect::<Vec<_>>() as _,
+            THRESHOLD
+        )
+        .fetch_optional(&self.pool)
+        .await
+        .map_err(Into::into)
+    }
+
+    async fn get_similar_side(
+        &self,
+        similar_name: &str,
+        allergens: &[Allergen],
+        additives: &[Additive],
+    ) -> Result<Option<Uuid>> {
+        sqlx::query_scalar!(
+            // the `<@` operator checks whether each element in the left array is also present in the right
+            "
+            SELECT food_id 
+            FROM food
+            WHERE similarity(name, $1) >= $4 AND food_id NOT IN (SELECT food_id FROM meal)
+            AND food_id IN (
+                SELECT food_id 
+                FROM food_allergen FULL JOIN food USING (food_id)
+                GROUP BY food_id 
+				HAVING COALESCE(array_agg(allergen) FILTER (WHERE allergen IS NOT NULL), ARRAY[]::allergen[]) <@ $2::allergen[] 
+				AND COALESCE(array_agg(allergen) FILTER (WHERE allergen IS NOT NULL), ARRAY[]::allergen[]) @> $2::allergen[]
+            )
+            AND food_id IN (
+                SELECT food_id
+				FROM food_additive FULL JOIN food USING (food_id)
+				GROUP BY food_id 
+				HAVING COALESCE(array_agg(additive) FILTER (WHERE additive IS NOT NULL), ARRAY[]::additive[]) <@ $3::additive[] 
+				AND COALESCE(array_agg(additive) FILTER (WHERE additive IS NOT NULL), ARRAY[]::additive[]) @> $3::additive[]
+            )
+            ORDER BY similarity(name, $1) DESC
+            ",
+            similar_name,
+            allergens
+                .iter()
+                .copied()
+                .map(Allergen::to_db_string)
+                .collect::<Vec<_>>() as _,
+            additives
+                .iter()
+                .copied()
+                .map(Additive::to_db_string)
+                .collect::<Vec<_>>() as _,
+            THRESHOLD
+        )
+        .fetch_optional(&self.pool)
+        .await
+        .map_err(Into::into)
+    }
+
+    async fn update_canteen(&self, uuid: Uuid, name: &str, position: u32) -> Result<Uuid> {
+        sqlx::query_scalar!(
+            "
+            UPDATE canteen
+            SET name = $2, position = $3
+            WHERE canteen_id = $1
+            RETURNING canteen_id
+            ",
+            uuid,
+            name,
+            i32::try_from(position)?
+        )
+        .fetch_one(&self.pool)
+        .await
+        .map_err(Into::into)
+    }
+
+    async fn update_line(&self, uuid: Uuid, name: &str, position: u32) -> Result<Uuid> {
+        sqlx::query_scalar!(
+            "
+            UPDATE line
+            SET name = $2, position = $3
```

Comment:
Canteens will be ordered by the env string. Lines could be a problem, if a line changes its canteen. We could add to similarity test, that the canteen_id must be similar. So each canteen has its own set of lines.
Good point.

Reply:
we cloud add a UNIQUE constrain to (canteen_id, position) in the database. But this is problematic as lines may change and then a new line could not be inserted (which is a case we then have to handle; how?). I think it is better thath when a line is replaced the app shows both lines at the same position instead of not showing the new line at all.